database/secret/update.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+//nolint:dupl // ignore similar code with create.go
+package secret
+
+import (
+	"fmt"
+
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// UpdateSecret updates an existing secret in the database.
+func (e *engine) UpdateSecret(s *library.Secret) error {
+	// handle the secret based off the type
+	switch s.GetType() {
+	case constants.SecretShared:
+		e.logger.WithFields(logrus.Fields{
+			"org":    s.GetOrg(),
+			"team":   s.GetTeam(),
+			"secret": s.GetName(),
+			"type":   s.GetType(),
+		}).Tracef("updating secret %s/%s/%s/%s in the database", s.GetType(), s.GetOrg(), s.GetTeam(), s.GetName())
+	default:
+		e.logger.WithFields(logrus.Fields{
+			"org":    s.GetOrg(),
+			"repo":   s.GetRepo(),
+			"secret": s.GetName(),
+			"type":   s.GetType(),
+		}).Tracef("updating secret %s/%s/%s/%s in the database", s.GetType(), s.GetOrg(), s.GetRepo(), s.GetName())
+	}
```

Comment:
Same as `Create`. Please add tests to ensure this can update all secret types: Org, Team, Repo.